### ISSUE TITLE
Confidence doesn’t compare valuee

### ISSUE THREAD

**Issue by rickmark** [Vector35/binaryninja-api#3869]
https://github.com/Vector35/binaryninja-api/blob/5b41ad47e381a8807e921977517c0108dfbf6367/binaryninjaapi.h#L456

I believe this line is intended to check values for equality , the next checks confidence

**Comment by ccarpenter04** [comment]
If this is indeed the case I would expect to see some pretty wide reaching effects in some cases.

**Comment by plafosse** [comment]
Yes it absolutely should be. The reason why it hasn't been a huge deal is that we use a separate implementation in the core
```
		bool operator==(const Confidence<T>& a) const
		{
			if (m_confidence != a.m_confidence)
				return false;
			return m_value == a.m_value;
		}
```

**Comment by negasora** [comment]
Fixed in e4ecd5be974b417f281916918492da2c32aa16e2
